scripts/other/mteb_meta.py

```diff
@@ -0,0 +1,109 @@
+"""
+Usage: python mteb_meta.py path_to_results_folder
+
+Creates evaluation results metadata for the model card. 
+E.g.
+---
+tags:
+- mteb
+model-index:
+- name: SGPT-5.8B-weightedmean-msmarco-specb-bitfit
+  results:
+  - task:
+      type: classification
+    dataset:
+      type: mteb/banking77
+      name: MTEB Banking77
+      config: default
+      split: test
+    metrics:
+    - type: accuracy
+      value: 84.49350649350649
+---
+"""
+
+import io
+import json
+import os
+import sys
+
+from mteb import MTEB
+
+results_folder = sys.argv[1].strip("/")
+model_name = results_folder.split("/")[-1]
+
+all_results = {}
+
+for file_name in os.listdir(results_folder):
+    if not file_name.endswith(".json"):
+        print(f"Skipping non-json {file_name}")
+        continue
+    with io.open(os.path.join(results_folder, file_name), "r", encoding="utf-8") as f:
+        results = json.load(f)
+        all_results = {**all_results, **{file_name.replace(".json", ""): results}}
+
+MARKER = "---"
+TAGS = "tags:"
+MTEB_TAG = "- mteb"
+HEADER = "model-index:"
+MODEL = f"- name: {model_name}"
+RES = "  results:"
+
+META_STRING = "\n".join([MARKER, TAGS, MTEB_TAG, HEADER, MODEL, RES])
+
+
+ONE_TASK = "  - task:\n      type: {}\n    dataset:\n      type: {}\n      name: {}\n      config: {}\n      split: {}\n    metrics:"
```

Comment:
`config` should probably be renamed `lang`?

Reply:
There is no lang - This refers to the dataset config, see https://github.com/huggingface/hub-docs/blame/main/modelcard.md

But you remind me that we should also add the revision kwarg 👍